clippy_lints/src/precedence.rs

```diff
@@ -86,11 +103,16 @@ impl EarlyLintPass for Precedence {
         }
 
         if let ExprKind::Unary(UnOp::Neg, ref rhs) = expr.kind {
-            if let ExprKind::MethodCall(_, ref args) = rhs.kind {
+            if let ExprKind::MethodCall(ref path_segment, ref args) = rhs.kind {
                 if let Some(slf) = args.first() {
                     if let ExprKind::Lit(ref lit) = slf.kind {
                         match lit.kind {
                             LitKind::Int(..) | LitKind::Float(..) => {
+                                for &odd_function in &ODD_FUNCTIONS_WHITELIST {
+                                    if odd_function == &*path_segment.ident.name.as_str() {
+                                        return;
+                                    }
+                                }
```

Comment:
I order to not compute `&*path_segment.ident.name.as_str()` for every `odd_function`, we can cache the result in a variable outside of the loop and reuse it:

````rust
....
           let path_seg_str = path_segment.ident.name.as_str(); // this is new
                if let Some(slf) = args.first() {
                    if let ExprKind::Lit(ref lit) = slf.kind {
                        match lit.kind {
                            LitKind::Int(..) | LitKind::Float(..) => {
                                if ODD_FUNCTIONS_WHITELIST
                                    .iter()
                                    .any(|odd_function| *odd_function == &*path_seg_str) // use here
                                {
                                    return;
                                };
````